test-suite/squidconf/dstdomain-domain-dup.conf

```diff
@@ -0,0 +1,8 @@
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+acl foo dstdomain .example.com .example.com
```

Comment:
Please make this test more complex. Something along these lines, perhaps?

```
acl foo dstdomain one.example.com two.example.com
acl foo dstdomain three.example.com one.example.co one.example.com
```

Same for the other tests, positive and negative -- make them as tricky as we can make them while still focusing on a single problem per test.